### PR TITLE
Hotfix/ia ol importbot

### PR DESCRIPTION
**Pull request by mekarpeles**
Flagged as work in progress, still requires testing but the logic is all together and most components have been tested.

### CODE REVIEW

File: openlibrary/core/ia.py [hunk 1]
```diff
@@ -292,3 +298,90 @@ def add_subjects(self):
...
+
+    Talk to archive.org database directly if it is specified in the
+    global configuration or if a configfile is provided.
+    """
+    if configfile:
+        from openlibrary.config import load_config
+        load_config(configfile)
+
+    if not config.get("ia_db"):
+        return
```
Comment (bfalling) [inline openlibrary/core/ia.py]: Would it be more clear to return `None`? Or does convention suggest not specifying a return value?

File: openlibrary/core/ia.py [hunk 2]
```diff
@@ -292,3 +298,90 @@ def add_subjects(self):
...
+        qvars['min_scandate'] = min_scandate.strftime("%Y%m%d")
+        q += " AND scandate > $min_scandate"
+
+    if date:
+        qvars['date'] = date.isoformat()
+        q += " AND updated > $date AND updated < ($date::date + INTERVAL '1' DAY)"
+
+    results = db.query(q, vars=qvars)
+    return results
+    return results if lazy else [row.identifier for row in results]
```
Comment (bfalling) [inline openlibrary/core/ia.py]: Should this line be removed, since it's never hit?

File: openlibrary/plugins/importapi/code.py [hunk 3]
```diff
@@ -224,17 +224,37 @@ def POST(self):
         # Case 4 - Does this item have a marc record?
         marc_record = self.get_marc_record(identifier)
         if not marc_record:
+            # return self.load_book(self.get_ia_record(metadata))
```
Comment (bfalling) [inline openlibrary/plugins/importapi/code.py]: Remove dead code, unless needed as a reference?

File: openlibrary/core/ia.py (hunk 2 above)
Comment (mekarpeles) [inline openlibrary/core/ia.py]: Ah sorry! that results line is definitely part of [WIP] -- line 386 needs to be removed. Thanks!

File: openlibrary/plugins/importapi/code.py (hunk 3 above)
Comment (mekarpeles) [inline openlibrary/plugins/importapi/code.py]: The import process has been changed to accept an option of requiring "marcs". When finished, if marcs=False (i.e. not required) I think we'll want to call this block.

For the interim, we can just remove it and punt on this feature.

File: openlibrary/core/ia.py (hunk 1 above)
Comment (mekarpeles) [inline openlibrary/core/ia.py]: Yes it would. legacy code (only moved from one place to another) -- will update according to suggestion

**Comment by mekarpeles** [comment]
@bfalling ping

File: scripts/manage-imports.py [hunk 4]
```diff
@@ -69,82 +103,84 @@ def add_new_scans(args):
...
             time.sleep(60)
 
         for item in items:
-            do_import(item)
+            do_import(item, servername=servername)
+
+def retroactive_import(start=None, stop=None, servername=None):
+    """Retroactively get all new scribe3 repub states (in all time)"""
+    items = get_candidate_ocaids(
+        scanned_within_days=None, repub_states=[19, 20, 22])[start:stop]
```
Comment (bfalling) [inline scripts/manage-imports.py]: Can we replace the magic `repub_states` with a named constant?

File: openlibrary/core/ia.py [hunk 5]
```diff
@@ -292,3 +298,99 @@ def add_subjects(self):
...
+        'c1': '%opensource%',
+        'c2': '%additional_collections%'
+    }
+
+    _valid_repub_states_sql = "(%s)" % (', '.join(str(i) for i in repub_states))
+    q = (
+        "SELECT " + ("count(identifier)" if count else "identifier") + " FROM metadata" +
+        " WHERE repub_state IN " + _valid_repub_states_sql +
+        "   AND mediatype='texts'" +
+        "   AND (noindex IS NULL OR collection='printdisabled')" +
```
Comment (bfalling) [inline openlibrary/core/ia.py]: Just checking: Does the `printdisabled` collection cover all the necessary cases? i.e. We don't need `lendinglibrary` or `inlibrary`? Is `printdisabled` the superset?

File: scripts/manage-imports.py (hunk 4 above)
Comment (mekarpeles) [inline scripts/manage-imports.py]: Sure thing, we can do this

File: openlibrary/core/ia.py (hunk 5 above)
Comment (mekarpeles) [inline openlibrary/core/ia.py]: Hm, I wondered this too. I talked to Alexis and she suggested that printdisabled is a superset of anything that would be inlibrary.

I remember asking about `lendinglibrary`, I think the answer was that they would not be no-index or would be in printdisabled as well. I can check w/ Alexis again.

**Comment by bfalling** [comment]
LGTM

**Pull request merged**